testutil/base.go

```diff
@@ -60,3 +62,31 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// Mocks any mutable object with a compliant substitute, and insures it will be restored after the test
+func (s *BaseTest) MockAny(mockedByPtr interface{}, mock interface{}) {
+	restore := MockAny(mockedByPtr, mock)
+	s.AddCleanup(restore)
+}
+
+// Mocks any mutable object with a compliant substitute, and returns a restore function
+func MockAny(mockedByPtr interface{}, mock interface{}) (restore func()) {
+	mockedPtr, mockValue := reflect.ValueOf(mockedByPtr), reflect.ValueOf(mock)
+
+	if mockedPtr.Type().Kind() != reflect.Ptr {
+		panic("Mock: first argument shall be a pointer to the mocked element!")
+	}
+
+	if mockedPtr.Elem().Type() != mockValue.Type() {
+		panic(fmt.Sprintf("Mock: mock type shall be '%s', but '%s' passed instead!", mockedPtr.Elem().Type(), mockValue.Type()))
```

Comment:
this is the part I like the least of the proposal, it's turning a compile time error into a panic. We have other test helpers that panic but in this case the trade off for the win is not clear.

Reply:
@pedronis, I see your point. When I tried to play with it, I saw that it doesn't fail in compile time, because of `reflect` library. All checkings took place in runtime, but maybe I missed something there.
However, when generics will eventually make it into official GoLang version and we will move to it, then it will become a compile time thing.